File: src/proficiency/math/prime_checker/PrimeChecker.java
import java.lang.Math;

public class PrimeChecker {
    public static boolean primeChecker(int n) {
        if (n < 2) {
            return false;
        } else if (n == 2) {
            return true;
        } else if (n % 2 == 0) {
            return false;
        } else {
            double lim = Math.ceil(Math.sqrt((double) n));
            for (int i=3; i<lim; i+=2) {
                if (n % i == 0) {
                    return false;
                }
            }
            return true;
        }
    }

    public static void main(String[] args) {
        int i = Integer.parseInt(args[0]);
        System.out.println(primeChecker(i));
    }
}

File: src/proficiency/math/prime_checker/prime_checker.cpp
// A very basic prime checker with minor optimizations
#include <iostream>
#include <string>
#include <cmath>

using namespace std;

bool prime_checker(int n) {
    if (n < 2) {
        return false;
    } else if (n == 2) {
        return true;
    } else if (n % 2 == 0) {
        return false;
    } else {
        double lim = ceil(sqrt(static_cast<double>(n)));
        for (int i=3; i<lim; i++) {
            if (n % i == 0) {
                return false;
            }
        }
        return true;
    }
}

int main(int argc, char* args[]) {
    int flag = prime_checker(stoi(args[1]));
    string is_prime = flag? "true" : "false";

    cout << is_prime << endl;
}
